pkg/receiver/rawk8seventsreceiver/receiver.go

```diff
@@ -103,22 +103,30 @@ func newRawK8sEventsReceiver(
 	restClient := client.CoreV1().RESTClient()
 	for _, namespace := range namespaces {
 		namespaceListWatch := listerWatcherFactory(restClient, "events", namespace, fields.Everything())
-		_, namespaceController = cache.NewInformer(namespaceListWatch, &corev1.Event{}, 0, cache.ResourceEventHandlerFuncs{
```

Comment:
```NewInformer``` in v0.31.1 deprecated: https://pkg.go.dev/k8s.io/client-go/tools/cache#NewInformer

changed to use ```InformerOptions```

Reply:
if this is deprecated, could this be removed in the next release? This could stay in this release, since there are already other changes in it.

Reply:
remove which? 🤔 ```NewInformer``` is removed already (deprecated)

Reply:
deprecated doesn't mean that its removed already. In other words, its not a breaking change yet.

Reply:
I'm trying to reduce the number of PR changes which potentially impact a feature.

Reply:
ohhh gotcha

Reply:
ah yeah i remember why this was included in the PR:

running: ```make golint```

```
running make lint in ./pkg/receiver/rawk8seventsreceiver
golangci-lint run
WARN [config_reader] The configuration option `linters.gosimple.go` is deprecated, please use global `run.go`. 
receiver.go:106:28: SA1019: cache.NewInformer is deprecated: Use NewInformerWithOptions instead. (staticcheck)
                _, namespaceController = cache.NewInformer(namespaceListWatch, &corev1.Event{}, 0, cache.ResourceEventHandlerFuncs{
                                         ^
make[2]: *** [lint] Error 1
make[1]: *** [for-all] Error 2
make: *** [golint] Error 2
```

Reply:
reverting ```client-go``` version to what it currently is won't work either since the repo need to run ```go mod tidy``` after which auto updates the client go package to 0.31

unless theres a hacky way to not update client go package?

Reply:
I think we ignore deprecation using
//lint:ignore SA1019 Need to keep deprecated package for compatibility.

Reply:
oh nice, figured it out added:

```// nolint:staticcheck```

didn't know about this for Go, learned something new!